freqtrade/freqai/prediction_models/ReinforcementLearner_multiproc.py

```diff
@@ -37,14 +37,14 @@ def set_train_and_eval_environments(self, data_dictionary: Dict[str, Any],
         env_id = "train_env"
         self.train_env = SubprocVecEnv([make_env(self.MyRLEnv, env_id, i, 1, train_df, prices_train,
                                         self.reward_params, self.CONV_WIDTH, monitor=True,
-                                        config=self.config) for i
+                                        config=self.config, dp=self.data_provider) for i
```

Comment:
are you sure this actually works? (especially looking at dry/live...).

in my understanding of python - creating a sub-process means that all objects need to be pickled, and will then be unpickled in the other process. Maybe SubprocVecEnv uses some different mechanics under the hood so it works - but standard python / joblib it will not.

this is impossible for objects with open sockets (open network connections) - which dataprovider will have implicitly by having an exchange attached to it.

This is untrue in hyperopt (for that very same reason) - so i

Reply:
It doesn't work yes, I am pushing another commit to fix it.

Reply:
Basically we are now passing an info dict to base environment for additional data, and the backtesting bug was because of the reset_env being called before setting add_state_info to false.